demo/src/main/java/com/novoda/storagepathfinder/demo/StoragePathsAdapter.kt

```diff
@@ -0,0 +1,29 @@
+package com.novoda.storagepathfinder.demo
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.novoda.storagepathfinder.StoragePath
+
+typealias Listener<T> = (T) -> Unit
+
+internal class StoragePathsAdapter(private val layoutInflater: LayoutInflater,
+                                   private val deviceStoragePaths: List<StoragePath>,
+                                   private val onAddFileClicked: Listener<StoragePath>) : RecyclerView.Adapter<StoragePathViewHolder>() {
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoragePathViewHolder =
+            StoragePathViewHolder(layoutInflater.inflate(R.layout.item_device_storage_root, parent, false))
```

Comment:
Why do we inject the `LayoutInflater`? Here we can get onw by using `LayoutInflater.from(parent.context)`

Also, here we can do a private extension like:
```
private fun ViewGroup.createItemView() = 
  LayoutInflater.from(context).inflate(R.layout.item_device_storage_root, this, false)
```
So we can then override `onCreateViewHolder` as:
```
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = parent.createItemView()
```
See there we don't really need to define the return value. I personally omit it when overriding functions since there super class has already defined it...